cli/azd/pkg/tools/dotnet/dotnet.go

```diff
@@ -178,18 +192,62 @@ func (cli *dotNetCli) PublishContainer(
 		"/t:PublishContainer",
 		fmt.Sprintf("-p:ContainerImageName=%s", imageName),
 		fmt.Sprintf("-p:ContainerRegistry=%s", server),
+		"--getProperty:GeneratedContainerConfiguration",
 	)
 
 	runArgs = runArgs.WithEnv([]string{
 		fmt.Sprintf("SDK_CONTAINER_REGISTRY_UNAME=%s", username),
 		fmt.Sprintf("SDK_CONTAINER_REGISTRY_PWORD=%s", password),
 	})
 
-	_, err := cli.commandRunner.Run(ctx, runArgs)
+	result, err := cli.commandRunner.Run(ctx, runArgs)
 	if err != nil {
-		return fmt.Errorf("dotnet publish on project '%s' failed: %w", project, err)
+		return 0, fmt.Errorf("dotnet publish on project '%s' failed: %w", project, err)
 	}
-	return nil
+
+	port, err := cli.getTargetPort(result.Stdout)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get dotnet target port: %w", err)
+	}
+
+	return port, nil
+}
+
+func (cli *dotNetCli) getTargetPort(result string) (int, error) {
+	var targetPorts []targetPort
+	var configOutput ResponseContainerConfiguration
+
+	if err := json.Unmarshal([]byte(result), &configOutput); err != nil {
+		return 0, fmt.Errorf("unmarshal dotnet configuration output '%s' failed: %w", result, err)
+	}
+	var exposedPortOutput []string
+	for key := range configOutput.Config.ExposedPorts {
+		exposedPortOutput = append(exposedPortOutput, key)
+	}
+
+	// exposedPortOutput format is <PORT_NUM>[/PORT_TYPE>]
+	for _, value := range exposedPortOutput {
+		split := strings.Split(value, "/")
+		if len(split) > 1 {
+			targetPorts = append(targetPorts, targetPort{port: split[0], protocol: split[1]})
+		} else {
+			// Provide a default tcp protocol if none is specified
+			targetPorts = append(targetPorts, targetPort{port: split[0], protocol: "tcp"})
+		}
+	}
+
+	// TODO Handle Target Port for multiple ports - return error says it is not supported
+	// return port[0].port, nil
+	if len(exposedPortOutput) < 1 {
+		return 0, fmt.Errorf(
+			"multiple dotnet port %s detected", targetPorts)
+	}
+
+	port, err := strconv.Atoi(targetPorts[0].port)
```

Comment:
this is the `Atoi` that I wonder if we really need...

Reply:
If this `Atoi` failed and we had not done it, we'd get some from ARM when we submitted the YAML later and I suspect the message would be worse than what we have here.  I like doing this validation early when possible.

Reply:
I'm fine with that.. It is a port number, so it is not likely to ever grow beyond what a go int can hold.
I was just thinking that doing a value passthrough would be the equivalente of doing it without azd. So the chances for error are increased by azd doing a conversion there.

Hypothetic future, similar to IP4 to IP6, if there is a new PORT convention where numbers become HEX and now you can have a port like `11AE`, azd would be the break point there.

But yeah, I won't make a deal of that hypothetic scenario, hahahah